### PR TITLE
Bulk file additions

### PR DESCRIPTION
**Pull request by ivukotic**

### CODE REVIEW

File: src/servicex_did_finder_lib/servicex_adaptor.py [hunk 1]
```diff
@@ -88,6 +94,32 @@ def put_file_add(self, file_info):
             self.logger.error(f'After {attempts} tries, failed to send ServiceX App a put_file '
                               f'message: {str(file_info)} - Ignoring error.')
 
+    def put_file_add_bulk(self, file_list):
+        for chunk in chunks(file_list, 20):
```
Comment (BenGalewsky) [inline src/servicex_did_finder_lib/servicex_adaptor.py:98]: Should there really be a magic number of 20 here in this method? Maybe it should be an argument to `put_file_add_bulk`?

File: src/servicex_did_finder_lib/servicex_adaptor.py [hunk 2]
```diff
@@ -88,6 +94,32 @@ def put_file_add(self, file_info):
...
+            success = False
+            attempts = 0
+            mesg = []
+            for fi in chunk:
+                mesg.append({
+                    "timestamp": datetime.now().isoformat(),
```
Comment (BenGalewsky) [inline src/servicex_did_finder_lib/servicex_adaptor.py:104]: Duplicated code... can this be refactored to avoid two places where the json is created

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
Please add more unit tests. This PR drops code coverage pretty heavily

File: src/servicex_did_finder_lib/servicex_adaptor.py (hunk 1 above)
Comment (ivukotic) [reply src/servicex_did_finder_lib/servicex_adaptor.py]: hm. my comment from VScode disappeared... Anyhow, this number is a compromise between json size and number or requests generated. Given that some special datasets can have 20+ replicas, to be on the safe side I selected this number but everything from 20-100 will be OK. Making it an argument would only push ball along. We could do something fancier like summing up message size but that would be a bit of overkill.

**Comment by ivukotic** [comment]
Added tests, coverage is back.

File: src/servicex_did_finder_lib/servicex_adaptor.py [hunk 3]
```diff
@@ -95,18 +98,12 @@ def put_file_add(self, file_info):
                               f'message: {str(file_info)} - Ignoring error.')
 
     def put_file_add_bulk(self, file_list):
-        for chunk in chunks(file_list, 20):
+        for chunk in chunks(file_list, 30):
```
[the remark this answers is not in the document]
Comment (ivukotic) [reply src/servicex_did_finder_lib/servicex_adaptor.py]: what do you want to do? limit the size of request in bytes?

[the remark this answers is not in the document]
Comment (ivukotic) [reply src/servicex_did_finder_lib/servicex_adaptor.py]: if so, what magic number of bytes would you prefer?

[the remark this answers is not in the document]
Comment (BenGalewsky) [reply src/servicex_did_finder_lib/servicex_adaptor.py]: Pass it in from the did finder implementation

[the remark this answers is not in the document]
Comment (BenGalewsky) [reply src/servicex_did_finder_lib/servicex_adaptor.py]: You can default to 20 or 30 or whatever

**Review by BenGalewsky — APPROVED** [review]
Thank you - this is a great addition

**Pull request merged**